### ISSUE TITLE
Ragnarok Mobile: Eternal Love

### ISSUE THREAD

**Issue by brunocalmon** [anbox/anbox#1028]
I've installed anbox on my Ubuntu and also installed "Google play services" to play "Ragnarok Mobile Eternal Love"

- I installed anbox using the getting started page.
- Google Play Services, I did using this tutorial https://linuxuprising.com/2018/07/anbox-how-to-install-google-play-store.html

After this I have tried two ways to get RO.

First I've tried diretly from google play store, but it is never installed after downloaded and just vanishes

Second I downloaded the apk available in the oficial ragnarok page, it was succesful installed, and the login has been done with success too, but after this, the touch stops working completly and I couldn't start the game.

**Comment by salvprest** [comment]
it's because the app expect a touch input,  and don't works with mouse input, I'm working on it, on my pc I transform the click in a touch, and with this, all the apps and games works properly, after some pull request to fix the touch on linux X11 systems I will make a pull request to fix this

**Comment by brunocalmon** [comment]
I see,

Could you teach me how to change click in touch, then I can make it by myself while the pr doesn't come. :D

**Comment by NomadMatt** [comment]
Also waiting, same issue... @salvprest if you could help us you'd be something like a god... 😆

Or just someone with the heaven's key... Thanks 👍

**Comment by salvprest** [comment]
you need to modify the input code in anbox and then compile the entire source in a snap, so if if you want i can make a branch on my repo with my changes, but if you wait a little, I want to make a pull request here, but only after that other things are fixed

**Comment by NomadMatt** [comment]
@salvprest  Since I'm NewBie I'll wait. Do you know when it can be done? Thanks for the answer, best regards

**Comment by salvprest** [comment]
for now I made a pull request to fix the touch screen on linux with X11, but it is incomplete, I'm trying to finish it, is not so simple unfortunately, after that I want to make another pull request, and after that I will push the pull request to translate the mouse input in a touch input, for now the problem with this change is that we lose the ability to use the mouse wheel, so I'm trying to workaround this problem

**Comment by willdowglas** [comment]
any progress yet? I am having the same issue.

**Comment by willdowglas** [comment]
I've tried to use the commits from @salvprest , but I couldn't get it to compile properly.

**Comment by alvelvis** [comment]
Have you found a way to solve this issue?

**Comment by willdowglas** [comment]
@alvelvis I found a solution by using genymotion's free version. You can easily install the google play, and play the game. It was the only solution I found so far to get ROM running on linux.

**Comment by WhyNotHugo** [comment]
Using an actual touch screen isn't working either. Is that a separate issue?

Has anyone had any experience using the game on a touchscreen?

**Comment by WhyNotHugo** [comment]
I managed to get it working fine. Performance is smooth as butter (and I've Intel video, BTW).

- I manually built `anbox` with the `-DENABLE_TOUCH_INPUT=ON` flag. Without this flag, mouse, trackpad and touchscreen are all passed on as clicks, but the game doesn't support clicks. With this flag, touchscreen events are passed on as touch events. You still can't use a mouse/touchpad though, so you need a touchscreen device (my laptop has a touchscreen, so that worked).
- The network needs to be manually enabled. I used the `anbox-bridge` package for ArchLinux, but it basically uses [this](https://raw.githubusercontent.com/anbox/anbox/master/scripts/anbox-bridge.sh) script.
- I had to use an android image with gapps (ArchLinux has a `anbox-image-gapps` package). You'll need Google Play services to log in with Google, not sure what happens if you used facebook.
- Install `Google Play Games` as well as the game via the Google Play Store.
- Run the game, HF.

**Comment by LiliumSancta** [comment]
Any update on this?

**Comment by WhyNotHugo** [comment]
@LiliumSancta I published clear instructions on how to get the game running. What sort of additional update are you expecting?

**Comment by alvelvis** [comment]
> You still can't use a mouse/touchpad though, so you need a touchscreen device (my laptop has a touchscreen, so that worked).

So for playing with mouse and keyboard, what should be the steps?

**Comment by LiliumSancta** [comment]
@WhyNotHugo I dont have touchscreen device, need mouse/Keyboard support.

**Comment by WhyNotHugo** [comment]
The keyboard works fine too (forgot to mention that!), pretty much the same way a bluetooth keyboard works on a phone.

As for mouse itself, looks like it's the game that does not support using a mouse, so the fix should probably come from the game developers themselves. Otherwise, wait until @salvprest patches are done. I honestly can't help in that since it's quite beyond me.

**Comment by hdk5** [comment]
I think I can't manually control my shipgirls in Azur Lane for the same reason. I hope @salvprest's patches would also fix this problem for me.

**Comment by Chrobak77** [comment]
Could anyone provide a bit more info about using -DENABLE_TOUCH_INPUT=ON ?
I do understand it is a cmake flag while building from source. But how do you implement this when you use snap  (edge) container?

**Comment by WhyNotHugo** [comment]
@Chrobak77 You probably want to look at `script/build-with-docker.sh` (which calls `scripts/clean-build.sh`), and then build the snap package using [these instructions](https://snapcraft.io/first-snap/pre-built/linux/build). (the `snapcraft.yaml` file is inside the `snap` directory).

I haven't done it myself; if building manually from source, building a snap package is probably more work that installing anbox natively.

**Comment by RodrigoMedeirosRS** [comment]
Hello @WhyNotHugo, i build the Anbox following your instrutctions, but don't solve the problem. I never build the Anbox before, so I think I'm doing something wrong. You can help me? Here is my steps:

$ git clone https://github.com/anbox/anbox.git
$ cd anbox
$ mkdir build
$ cd build
$ cmake -DENABLE_TOUCH_INPUT=ON ..
$ make
$ sudo make install

At this point the ubuntu tell me the build are done, and the file as put into: /usr/local/anbox

To run i use the following command: anbox launch --package=org.anbox.appmgr --component=org.anbox.appmgr.AppViewActivity

When i start the game he have the same issue.

**Comment by YuriCalistrato** [comment]
Any update about this? Have some friends that would love to use anbox as a gaming alternative emulator. Waiting feedback to try it again. Ty the effort of making this possible! :clap:

**Comment by present86** [comment]
Is there any update?

**Comment by YuriCalistrato** [comment]
Waiting for this too...

**Comment by salvprest** [comment]
this should be solved now!

**Comment by windusasongko** [comment]
Is there any update? i still has this problem.

**Comment by mohammadanton** [comment]
> this should be solved now!

how can this be solved?
after I installl anbox with snap, install play store then install ragnarok mobile.
they game instanly freeze / blank

**Comment by juliolvfilho** [comment]
Now it works guys!
Running on Ubuntu 18.04 LTS perfectly!

```sh
sudo snap install --edge --devmode anbox
```

And then install Google Play Services
https://www.linuxuprising.com/2018/07/anbox-how-to-install-google-play-store.html

I'm using the review 177. Check the latest version https://build.snapcraft.io/user/anbox/anbox

**Comment by tobia** [comment]
@juliolvfilho I still get only a black window. I installed the latest snap from edge (183) then houdini + gapps, permissions + login into play store, and then I sideloaded the apk. It install successfully, but I only get the black window. What did you do different?

**Comment by belkevglaz** [comment]
I've the same issue with black screen on Ubuntu 18.04 and anbox version from snap. 
What information can be useful for try to solve?

**Comment by belkevglaz** [comment]
I still continue to try to launch ROM under anbox.
I tried several installations (snap, from source).

I have installed anbox from source commit 56c25f1 and google services after that. 
Collected some android logs and system-info output and based on them I can suppose that is a problem of black screen in OPenGL area. I suppose that root of cause in this error :
```
01-15 15:24:50.102  1097  1132 E EGL_emulation: tid 1132: eglCreateContext(919): error 0x3005 (EGL_BAD_CONFIG)
01-15 15:24:50.107  1097  1132 E EGL_emulation: [getAttribValue] Bad attribute idx
01-15 15:24:50.107  1097  1132 E EGL_emulation: tid 1132: eglGetConfigAttrib(613): error 0x3004 (EGL_BAD_ATTRIBUTE)
01-15 15:24:50.107  1097  1132 E EGL_emulation: [getAttribValue] Bad attribute idx
01-15 15:24:50.107  1097  1132 E EGL_emulation: tid 1132: eglGetConfigAttrib(613): error 0x3004 (EGL_BAD_ATTRIBUTE)
01-15 15:24:50.107  1097  1132 E EGL_emulation: [getAttribValue] Bad attribute idx
01-15 15:24:50.107  1097  1132 E EGL_emulation: tid 1132: eglGetConfigAttrib(613): error 0x3004 (EGL_BAD_ATTRIBUTE)
01-15 15:24:50.107  1097  1132 E EGL_emulation: tid 1132: eglCreateContext(919): error 0x3005 (EGL_BAD_CONFIG)
01-15 15:24:50.107  1097  1132 E Unity   : [EGL] Unable to create context!: EGL_BAD_CONFIG: An EGLConfig argument does not name a valid EGL frame buffer configuration.
01-15 15:24:50.107  1097  1132 E Unity   :  
01-15 15:24:50.107  1097  1132 E Unity   : (Filename: /Users/builduser/buildslave/unity/build/Runtime/GfxDevice/egl/ContextEGL.cpp Line: 111)
01-15 15:24:50.107  1097  1132 E Unity   : 
01-15 15:24:50.116  1097  1132 E EGL_emulation: tid 1132: eglCreateContext(919): error 0x3005 (EGL_BAD_CONFIG)
01-15 15:24:50.116  1097  1132 E Unity   : [EGL] Unable to create context!: EGL_BAD_CONFIG: An EGLConfig argument does not name a valid EGL frame buffer configuration.
01-15 15:24:50.116  1097  1132 E Unity   :  
01-15 15:24:50.116  1097  1132 E Unity   : (Filename: /Users/builduser/buildslave/unity/build/Runtime/GfxDevice/egl/ContextEGL.cpp Line: 111)
01-15 15:24:50.116  1097  1132 E Unity   : 
```
Could any help to fix it ? Additionally logs below.
<details>
<summary>system-info output</summary>
<p>

```
version: local-56c25f1
cpu:
  arch:  x86
  brand: Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
  features: 
    - aes
    - sse4_1
    - sse4_2
    - avx
    - avx2
os:
  name: Ubuntu
  version: 19.04 (Disco Dingo)
  snap-based: false
kernel:
  version: Linux version 5.0.0-38-generic (buildd@lgw01-amd64-036) (gcc version 8.3.0 (Ubuntu 8.3.0-6ubuntu1)) #41-Ubuntu SMP Tue Dec 3 00:27:35 UTC 2019
  binder: true
  ashmem: true
graphics:
  egl:
    vendor: NVIDIA
    version: 1.5
    extensions:
      - EGL_EXT_buffer_age
      - EGL_EXT_client_sync
      - EGL_EXT_create_context_robustness
      - EGL_EXT_output_base
      - EGL_EXT_stream_acquire_mode
      - EGL_EXT_sync_reuse
      - EGL_IMG_context_priority
      - EGL_KHR_config_attribs
      - EGL_KHR_create_context_no_error
      - EGL_KHR_context_flush_control
      - EGL_KHR_create_context
      - EGL_KHR_display_reference
      - EGL_KHR_fence_sync
      - EGL_KHR_get_all_proc_addresses
      - EGL_KHR_partial_update
      - EGL_KHR_swap_buffers_with_damage
      - EGL_KHR_no_config_context
      - EGL_KHR_gl_colorspace
      - EGL_KHR_gl_renderbuffer_image
      - EGL_KHR_gl_texture_2D_image
      - EGL_KHR_gl_texture_3D_image
      - EGL_KHR_gl_texture_cubemap_image
      - EGL_KHR_image
      - EGL_KHR_image_base
      - EGL_KHR_image_pixmap
      - EGL_KHR_reusable_sync
      - EGL_KHR_stream
      - EGL_KHR_stream_attrib
      - EGL_KHR_stream_consumer_gltexture
      - EGL_KHR_stream_cross_process_fd
      - EGL_KHR_stream_fifo
      - EGL_KHR_stream_producer_eglsurface
      - EGL_KHR_surfaceless_context
      - EGL_KHR_wait_sync
      - EGL_NV_nvrm_fence_sync
      - EGL_NV_post_sub_buffer
      - EGL_NV_stream_cross_display
      - EGL_NV_stream_cross_object
      - EGL_NV_stream_cross_process
      - EGL_NV_stream_cross_system
      - EGL_NV_stream_flush
      - EGL_NV_stream_metadata
      - EGL_NV_stream_remote
      - EGL_NV_stream_reset
      - EGL_NV_stream_socket
      - EGL_NV_stream_socket_inet
      - EGL_NV_stream_socket_unix
      - EGL_NV_stream_sync
      - EGL_NV_stream_fifo_next
      - EGL_NV_stream_fifo_synchronous
      - EGL_NV_stream_consumer_gltexture_yuv
      - EGL_NV_stream_attrib
      - EGL_NV_system_time
      - EGL_NV_output_drm_flip_event
  gles2:
    vendor: NVIDIA Corporation
    vendor: OpenGL ES 3.2 NVIDIA 418.56
    extensions:
      - GL_EXT_base_instance
      - GL_EXT_blend_func_extended
      - GL_EXT_blend_minmax
      - GL_EXT_buffer_storage
      - GL_EXT_clear_texture
      - GL_EXT_clip_control
      - GL_EXT_clip_cull_distance
      - GL_EXT_color_buffer_float
      - GL_EXT_color_buffer_half_float
      - GL_EXT_conservative_depth
      - GL_EXT_copy_image
      - GL_EXT_debug_label
      - GL_EXT_discard_framebuffer
      - GL_EXT_disjoint_timer_query
      - GL_EXT_draw_buffers_indexed
      - GL_EXT_draw_elements_base_vertex
      - GL_EXT_EGL_image_array
      - GL_EXT_EGL_image_storage
      - GL_EXT_EGL_image_external_wrap_modes
      - GL_EXT_float_blend
      - GL_EXT_frag_depth
      - GL_EXT_geometry_point_size
      - GL_EXT_geometry_shader
      - GL_EXT_gpu_shader5
      - GL_EXT_map_buffer_range
      - GL_EXT_multi_draw_indirect
      - GL_EXT_multisample_compatibility
      - GL_EXT_multisampled_render_to_texture
      - GL_EXT_multisampled_render_to_texture2
      - GL_EXT_occlusion_query_boolean
      - GL_EXT_polygon_offset_clamp
      - GL_EXT_post_depth_coverage
      - GL_EXT_primitive_bounding_box
      - GL_EXT_raster_multisample
      - GL_EXT_render_snorm
      - GL_EXT_robustness
      - GL_EXT_separate_shader_objects
      - GL_EXT_shader_group_vote
      - GL_EXT_shader_implicit_conversions
      - GL_EXT_shader_integer_mix
      - GL_EXT_shader_io_blocks
      - GL_EXT_shader_non_constant_global_initializers
      - GL_EXT_shader_texture_lod
      - GL_EXT_shadow_samplers
      - GL_EXT_sparse_texture
      - GL_EXT_sparse_texture2
      - GL_EXT_sRGB
      - GL_EXT_sRGB_write_control
      - GL_EXT_tessellation_point_size
      - GL_EXT_tessellation_shader
      - GL_EXT_texture_border_clamp
      - GL_EXT_texture_buffer
      - GL_EXT_texture_compression_bptc
      - GL_EXT_texture_compression_dxt1
      - GL_EXT_texture_compression_rgtc
      - GL_EXT_texture_compression_s3tc
      - GL_EXT_texture_cube_map_array
      - GL_EXT_texture_filter_anisotropic
      - GL_EXT_texture_filter_minmax
      - GL_EXT_texture_format_BGRA8888
      - GL_EXT_texture_mirror_clamp_to_edge
      - GL_EXT_texture_norm16
      - GL_EXT_texture_rg
      - GL_EXT_texture_sRGB_R8
      - GL_EXT_texture_sRGB_decode
      - GL_EXT_texture_storage
      - GL_EXT_texture_view
      - GL_EXT_draw_transform_feedback
      - GL_EXT_unpack_subimage
      - GL_EXT_window_rectangles
      - GL_KHR_context_flush_control
      - GL_KHR_debug
      - GL_EXT_memory_object
      - GL_EXT_memory_object_fd
      - GL_KHR_parallel_shader_compile
      - GL_KHR_no_error
      - GL_KHR_robust_buffer_access_behavior
      - GL_KHR_robustness
      - GL_EXT_semaphore
      - GL_EXT_semaphore_fd
      - GL_NV_bgr
      - GL_NV_bindless_texture
      - GL_NV_blend_equation_advanced
      - GL_NV_blend_equation_advanced_coherent
      - GL_NVX_blend_equation_advanced_multi_draw_buffers
      - GL_NV_blend_minmax_factor
      - GL_NV_clip_space_w_scaling
      - GL_NV_conditional_render
      - GL_NV_conservative_raster
      - GL_NV_conservative_raster_pre_snap_triangles
      - GL_NV_copy_buffer
      - GL_NV_copy_image
      - GL_NV_draw_buffers
      - GL_NV_draw_instanced
      - GL_NV_draw_texture
      - GL_NV_draw_vulkan_image
      - GL_NV_EGL_stream_consumer_external
      - GL_NV_explicit_attrib_location
      - GL_NV_fbo_color_attachments
      - GL_NV_fill_rectangle
      - GL_NV_fragment_coverage_to_color
      - GL_NV_fragment_shader_interlock
      - GL_NV_framebuffer_blit
      - GL_NV_framebuffer_mixed_samples
      - GL_NV_framebuffer_multisample
      - GL_NV_generate_mipmap_sRGB
      - GL_NV_geometry_shader_passthrough
      - GL_NV_instanced_arrays
      - GL_NV_internalformat_sample_query
      - GL_NV_gpu_shader5
      - GL_NV_image_formats
      - GL_NV_memory_attachment
      - GL_NV_occlusion_query_samples
      - GL_NV_non_square_matrices
      - GL_NV_pack_subimage
      - GL_NV_packed_float
      - GL_NV_packed_float_linear
      - GL_NV_path_rendering
      - GL_NV_path_rendering_shared_edge
      - GL_NV_pixel_buffer_object
      - GL_NV_polygon_mode
      - GL_NV_read_buffer
      - GL_NV_read_depth
      - GL_NV_read_depth_stencil
      - GL_NV_read_stencil
      - GL_NV_sample_locations
      - GL_NV_sample_mask_override_coverage
      - GL_NV_shader_atomic_fp16_vector
      - GL_NV_shader_noperspective_interpolation
      - GL_NV_shadow_samplers_array
      - GL_NV_shadow_samplers_cube
      - GL_NV_sRGB_formats
      - GL_NV_stereo_view_rendering
      - GL_NV_texture_array
      - GL_NV_texture_barrier
      - GL_NV_texture_border_clamp
      - GL_NV_texture_compression_latc
      - GL_NV_texture_compression_s3tc
      - GL_NV_texture_compression_s3tc_update
      - GL_NV_timer_query
      - GL_NV_viewport_array
      - GL_NV_viewport_array2
      - GL_NV_viewport_swizzle
      - GL_KHR_blend_equation_advanced
      - GL_KHR_blend_equation_advanced_coherent
      - GL_OES_compressed_ETC1_RGB8_texture
      - GL_EXT_compressed_ETC1_RGB8_sub_texture
      - GL_OES_depth24
      - GL_OES_depth32
      - GL_OES_depth_texture
      - GL_OES_depth_texture_cube_map
      - GL_OES_copy_image
      - GL_OES_draw_buffers_indexed
      - GL_OES_draw_elements_base_vertex
      - GL_OES_texture_border_clamp
      - GL_OES_tessellation_point_size
      - GL_OES_tessellation_shader
      - GL_OES_texture_buffer
      - GL_OES_geometry_point_size
      - GL_OES_geometry_shader
      - GL_OES_gpu_shader5
      - GL_OES_shader_io_blocks
      - GL_OES_texture_view
      - GL_OES_primitive_bounding_box
      - GL_OES_EGL_image
      - GL_OES_EGL_image_external
      - GL_OES_EGL_image_external_essl3
      - GL_OES_EGL_sync
      - GL_OES_element_index_uint
      - GL_OES_fbo_render_mipmap
      - GL_OES_get_program_binary
      - GL_OES_mapbuffer
      - GL_OES_packed_depth_stencil
      - GL_OES_rgb8_rgba8
      - GL_OES_sample_shading
      - GL_OES_sample_variables
      - GL_OES_shader_image_atomic
      - GL_OES_shader_multisample_interpolation
      - GL_OES_standard_derivatives
      - GL_OES_surfaceless_context
      - GL_OES_texture_cube_map_array
      - GL_OES_texture_npot
      - GL_OES_texture_float
      - GL_OES_texture_float_linear
      - GL_OES_texture_half_float
      - GL_OES_texture_half_float_linear
      - GL_OES_texture_stencil8
      - GL_OES_texture_storage_multisample_2d_array
      - GL_OES_vertex_array_object
      - GL_OES_vertex_half_float
      - GL_OES_viewport_array
      - GL_OVR_multiview
      - GL_OVR_multiview2
      - GL_OVR_multiview_multisampled_render_to_texture
      - GL_ANDROID_extension_pack_es31a
```
</p>
</details> 

<details>
<summary>Part of android system.log </summary>

```
01-15 15:24:48.817  1097  1097 D MSC_LOG : QMSPLogin Begin
01-15 15:24:48.943  1097  1097 D MSC_LOG : QMSPLogin End
01-15 15:24:48.943  1097  1097 D MscSpeechLog: [MSPLogin]ret:0
01-15 15:24:48.969  1097  1097 D MscSpeechLog: DC init enable=null
01-15 15:24:48.969  1097  1097 D MscSpeechLog: DC init enter.
01-15 15:24:48.969  1097  1097 D MscSpeechLog: DC getEnable enter.
01-15 15:24:48.969  1097  1097 D MscSpeechLog: DC getEnable static value=true
01-15 15:24:48.969  1097  1097 D MscSpeechLog: getMscParameter enter key=idc
01-15 15:24:48.969  1097  1097 D MscSpeechLog: getMscParameter MSC return -1
01-15 15:24:48.969  1097  1097 D MscSpeechLog: getMscParameter value=null
01-15 15:24:48.969  1097  1097 D MscSpeechLog: getMscParameter leave
01-15 15:24:48.969  1097  1097 D MscSpeechLog: DC getEnable msc val=null
01-15 15:24:48.969  1097  1097 D MscSpeechLog: DC getEnable value=false
01-15 15:24:48.970  1097  1097 D MscSpeechLog: DC getEnable leave.
01-15 15:24:48.970  1097  1097 D MscSpeechLog: DC init is not enable.
01-15 15:24:48.970  1097  1097 D MscSpeechLog: DC init leave.
01-15 15:24:48.972  1097  1147 D MscSpeechLog: MscHttpRequest start Post
01-15 15:24:48.975  1097  1146 D UserLogger:  start mergerWifiList
01-15 15:24:48.977  1097  1151 D MscSpeechLog: MscHttpRequest start Post
01-15 15:24:48.978  1097  1116 I FA      : Tag Manager is not found and thus will not be used
01-15 15:24:48.978  1097  1097 V OverSeaActivity: onResume called
01-15 15:24:49.057  1097  1097 D UnitySpeechRecognizer: SpeechRecognizer init() code = 0
01-15 15:24:49.078  1097  1154 D         : HostConnection::get() New Host Connection established 0xd7b86120, tid 1154
01-15 15:24:49.081  1097  1154 I OpenGLRenderer: Initialized EGL, version 1.4
01-15 15:24:49.082  1097  1154 D OpenGLRenderer: Swap behavior 1
01-15 15:24:49.087  1097  1126 W System  : ClassLoader referenced unknown path: /system/priv-app/PrebuiltGmsCore/lib/x86
01-15 15:24:49.088  1097  1126 D ApplicationLoaders: ignored Vulkan layer search path /system/priv-app/PrebuiltGmsCore/lib/x86:/system/priv-app/PrebuiltGmsCore/PrebuiltGmsCore.apk!/lib/x86:/system/lib:/vendor/lib for namespace 0xf7c85090
01-15 15:24:49.309   111   132 I ActivityManager: Displayed com.gravity.romg/.WWActivity: +842ms (total +1s954ms)
01-15 15:24:49.314  1097  1126 W AnalyticsUserIDStore: initStore should have been called before calling setUserID
01-15 15:24:49.315  1097  1126 W UserDataStore: initStore should have been called before calling setUserID
01-15 15:24:49.395  1097  1132 I Unity   : SystemInfo CPU = ARMv7 VFPv3 NEON VMH, Cores = 8, Memory = 15761mb
01-15 15:24:49.395  1097  1132 I Unity   : SystemInfo ARM big.LITTLE configuration: 4 big (mask: 240), 4 little (mask: 15)
01-15 15:24:49.396  1097  1132 I Unity   : ApplicationInfo com.gravity.romg version 1.0.10 build f5c9fc13-b197-4bee-a736-4731998a172f
01-15 15:24:49.396  1097  1132 I Unity   : Built from '2017.4/staging' branch, Version '2017.4.30f1 (c6fa43736cae)', Build type 'Release', Scripting Backend 'il2cpp'
01-15 15:24:49.663  1097  1132 D houdini : [1132] Added shared library /data/app/com.gravity.romg-1/lib/arm/libil2cpp.so for ClassLoader by Native Bridge.
01-15 15:24:50.100  1097  1132 D         : HostConnection::get() New Host Connection established 0xd1a10640, tid 1132
01-15 15:24:50.102  1097  1132 E EGL_emulation: tid 1132: eglCreateContext(919): error 0x3005 (EGL_BAD_CONFIG)
01-15 15:24:50.107  1097  1132 E EGL_emulation: [getAttribValue] Bad attribute idx
01-15 15:24:50.107  1097  1132 E EGL_emulation: tid 1132: eglGetConfigAttrib(613): error 0x3004 (EGL_BAD_ATTRIBUTE)
01-15 15:24:50.107  1097  1132 E EGL_emulation: [getAttribValue] Bad attribute idx
01-15 15:24:50.107  1097  1132 E EGL_emulation: tid 1132: eglGetConfigAttrib(613): error 0x3004 (EGL_BAD_ATTRIBUTE)
01-15 15:24:50.107  1097  1132 E EGL_emulation: [getAttribValue] Bad attribute idx
01-15 15:24:50.107  1097  1132 E EGL_emulation: tid 1132: eglGetConfigAttrib(613): error 0x3004 (EGL_BAD_ATTRIBUTE)
01-15 15:24:50.107  1097  1132 E EGL_emulation: tid 1132: eglCreateContext(919): error 0x3005 (EGL_BAD_CONFIG)
01-15 15:24:50.107  1097  1132 E Unity   : [EGL] Unable to create context!: EGL_BAD_CONFIG: An EGLConfig argument does not name a valid EGL frame buffer configuration.
01-15 15:24:50.107  1097  1132 E Unity   :  
01-15 15:24:50.107  1097  1132 E Unity   : (Filename: /Users/builduser/buildslave/unity/build/Runtime/GfxDevice/egl/ContextEGL.cpp Line: 111)
01-15 15:24:50.107  1097  1132 E Unity   : 
01-15 15:24:50.116  1097  1132 E EGL_emulation: tid 1132: eglCreateContext(919): error 0x3005 (EGL_BAD_CONFIG)
01-15 15:24:50.116  1097  1132 E Unity   : [EGL] Unable to create context!: EGL_BAD_CONFIG: An EGLConfig argument does not name a valid EGL frame buffer configuration.
01-15 15:24:50.116  1097  1132 E Unity   :  
01-15 15:24:50.116  1097  1132 E Unity   : (Filename: /Users/builduser/buildslave/unity/build/Runtime/GfxDevice/egl/ContextEGL.cpp Line: 111)
01-15 15:24:50.116  1097  1132 E Unity   : 
01-15 15:24:50.189  1097  1132 D Unity   : GL_EXT_debug_marker GL_OES_compressed_ETC1_RGB8_texture GL_OES_depth24 GL_OES_depth32 GL_OES_depth_texture GL_OES_EGL_image GL_OES_EGL_image_external GL_OES_element_index_uint GL_OES_packed_depth_stencil GL_OES_rgb8_rgba8 GL_OES_standard_derivatives GL_OES_texture_npot GL_OES_texture_float GL_OES_texture_float_linear GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_OES_vertex_half_float
01-15 15:24:50.200  1097  1167 D         : HostConnection::get() New Host Connection established 0xd7b87200, tid 1167
01-15 15:24:50.358  1097  1188 D AudioTrack: Client defaulted notificationFrames to 682 for frameCount 2048
01-15 15:24:50.613    32    93 D AudioFlinger: mixer(0xf6903180) throttle end: throttle time(154)
01-15 15:24:50.765    32    93 I AudioFlinger: BUFFER TIMEOUT: remove(4096) from active list on thread 0xf6903180
01-15 15:24:50.766   111   146 W InputDispatcher: channel '42268e6 com.gravity.romg/com.gravity.romg.WWActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
01-15 15:24:50.766   111   146 E InputDispatcher: channel '42268e6 com.gravity.romg/com.gravity.romg.WWActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
01-15 15:24:50.767   111   329 I ActivityManager: Process com.gravity.romg (pid 1097) has died
01-15 15:24:50.767   111   425 D GraphicsStats: Buffer count: 3
01-15 15:24:50.767   111   845 I WindowManager: WIN DEATH: Window{42268e6 u0 com.gravity.romg/com.gravity.romg.WWActivity}
01-15 15:24:50.767   111   845 W InputDispatcher: Attempted to unregister already unregistered input channel '42268e6 com.gravity.romg/com.gravity.romg.WWActivity (server)'
01-15 15:24:50.767   111   329 D ActivityManager: cleanUpApplicationRecord -- 1097
01-15 15:24:50.767   111   845 W WindowManager: Force-removing child win Window{51566b1 u0 SurfaceView - com.gravity.romg/com.gravity.romg.WWActivity} from container Window{42268e6 u0 com.gravity.romg/com.gravity.romg.WWActivity}
01-15 15:24:50.767    31    31 I Zygote  : Process 1097 exited due to signal (11)
01-15 15:24:50.767    31    31 I Zygote  : Process 1097 dumped core.
01-15 15:24:50.768   111   329 W ActivityManager: Force removing ActivityRecord{9ee0636 u0 com.gravity.romg/.WWActivity t4}: app died, no saved state
01-15 15:24:50.769   111   122 W WindowManager: Failed looking up window
01-15 15:24:50.769   111   122 W WindowManager: java.lang.IllegalArgumentException: Requested window android.os.BinderProxy@680ee27 does not exist
01-15 15:24:50.769   111   122 W WindowManager: 	at com.android.server.wm.WindowManagerService.windowForClientLocked(WindowManagerService.java:9425)
01-15 15:24:50.769   111   122 W WindowManager: 	at com.android.server.wm.WindowManagerService.windowForClientLocked(WindowManagerService.java:9416)
01-15 15:24:50.769   111   122 W WindowManager: 	at com.android.server.wm.WindowState$DeathRecipient.binderDied(WindowState.java:1800)
01-15 15:24:50.769   111   122 W WindowManager: 	at android.os.BinderProxy.sendDeathNotice(Binder.java:688)
01-15 15:24:50.770   111   122 I WindowManager: WIN DEATH: null
01-15 15:24:50.770   111   329 I AnboxPlatformServiceProxy: Task removed id=4
01-15 15:24:51.936   549   585 D InitAlarmsService: Clearing and rescheduling alarms.
01-15 15:24:52.772    32    93 W AudioFlinger: write blocked for 753 msecs, 1 delayed writes, thread 0xf6903180
01-15 15:24:53.368    41  1128 W SocketClient: write error (Broken pipe)
01-15 15:24:53.368    41  1156 W SocketClient: write error (Broken pipe)
01-15 15:24:53.368    41  1160 W SocketClient: write error (Broken pipe)
01-15 15:24:53.368    41  1155 W SocketClient: write error (Broken pipe)
01-15 15:24:53.368    41  1156 W DnsProxyListener: Error writing DNS result to client
01-15 15:24:53.368    41  1128 W DnsProxyListener: Error writing DNS result to client
01-15 15:24:53.368    41  1160 W DnsProxyListener: Error writing DNS result to client
01-15 15:24:53.368    41  1155 W DnsProxyListener: Error writing DNS result to client
01-15 15:24:54.002    41  1149 W SocketClient: write error (Broken pipe)
01-15 15:24:54.002    41  1149 W DnsProxyListener: Error writing DNS result to client
01-15 15:24:55.357    41  1152 W SocketClient: write error (Broken pipe)
01-15 15:24:55.357    41  1152 W DnsProxyListener: Error writing DNS result to client
01-15 15:25:01.074   111   128 E BatteryStatsService: power: Missing API
01-15 15:25:01.075   111   125 E KernelCpuSpeedReader: Failed to read cpu-freq: /sys/devices/system/cpu/cpu0/cpufreq/stats/time_in_state (No such file or directory)
01-15 15:25:01.076   111   125 E KernelUidCpuTimeReader: Failed to read uid_cputime: /proc/uid_cputime/show_uid_stat (No such file or directory)
01-15 15:25:23.137   677   694 I Finsky  : [91] kik.run(3): Stats for Executor: BlockingExecutor kke@2daf469[Running, pool size = 3, active threads = 0, queued tasks = 0, completed tasks = 11]
01-15 15:25:23.144   677   694 I Finsky  : [91] kik.run(3): Stats for Executor: LightweightExecutor kke@fe83eee[Running, pool size = 4, active threads = 0, queued tasks = 0, completed tasks = 73]
01-15 15:25:23.222   677   694 I Finsky  : [91] kik.run(3): Stats for Executor: bgExecutor kke@3237f8f[Running, pool size = 4, active threads = 0, queued tasks = 0, completed
```
</details>

**Comment by vinnyfs89** [comment]
Solved!

Using genymotion it was solved! :tada: :wink:

I published the solution with step by step how to run the game on Linux [here](https://github.com/vinnyfs89/ragnarok-m-on-linux) 

Hope it helps! :+1:

**Comment by belkevglaz** [comment]
Genymotion is good, but it works slower than anbox

**Comment by mtlcrusher** [comment]
i need help, i installed anbox in snap and trying to run ragnarok mobile.. but it only shows black screen for a while then it closed
